src/event-bridge/event-bus.ts

```diff
@@ -374,3 +419,75 @@ class ImportedEventBus<E extends EventBusRuleInput> extends EventBusBase<E> {
     super(bus);
   }
 }
+
+/**
+ * Defines an integration that can be used in the `pipe` function of an {@link EventBus} (Rule or Transform).
+ *
+ * ```ts
+ * const myEbIntegration = { // an Integration
+ *    kind: 'myEbIntegration',
+ *    eventBus: {
+ *       target: (props, targetInput) => {
+ *           // return a RuleTarget
+ *       }
+ *    }
+ * }
+ *
+ * new EventBus().when(() => true).pipe(myEbIntegration);
+ * ```
+ */
+export interface EventBusTargetIntegration<
+  // the payload type we expect to be transformed into before making this call.
+  P,
+  Props extends object | undefined = undefined
+> {
+  __payloadBrand: P;
```

Comment:
Document why you need this. `target` has no relation to `P`, so you store this here to carry the `P` type forward and ensure data integrity? Where are you using this to type check? Can you link to that code with `{@link}`?

When we use brands like this, where there is actually no value present even though the types suggests otherwise, we should always provide a comment.